docs/advanced_topics/icons.rst

```diff
@@ -0,0 +1,104 @@
+.. _icons:
+
+=====
+Icons
+=====
+
+Wagtail comes with an icon set.
+The icons are used throughout the admin interface.
+
+Elements that use icons are:
+
+- :ref:`Various menu items <admin_hooks>`
+- :ref:`ModelAdmin menu items <modeladmin_menu_icon>`
+- :ref:`Streamfield blocks <basic_block_type_icon>`
+- :ref:`Rich text editor toolbar buttons <extending_the_draftail_editor>`
+- ...
+
+This document describes how to choose and add icons.
+
+Available icons and their names
+-------------------------------
+
+Enable the :ref:`styleguide <styleguide>` to view icons and their names.
+
+Alternatively inspect the source code ``wagtail/admin/wagtail_hooks.py``.
+The filename without ``.svg`` is the icon name.
+
+Add a custom icon
+-----------------
+
+Draw or download an icon.
+
+The SVG should:
+
+ - Have a size of 16x16 points
+ - Contain a ``symbol`` tag
+ - Have ``id="icon-<name>"`` attribute
+ - Not set the ``fill`` attribute
+ - Not set the ``xmlns`` attribute
+
+This SVG:
+
+.. code-block:: html
+
+    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16">
+      <path fill="black" d="M0 0h16v16H0z"/>
+    </svg>
+
+
+Converted to a SVG symbol:
+
+.. code-block:: html
+
+    <symbol id="icon-square" viewBox="0 0 16 16">
+        <path d="M0 0h16v16H0z"/>
+    </symbol>
+
+
+Add the icon to the set with the ``register_icons`` hook.
+
+.. code-block:: python
+
+    @hooks.register("register_icons")
```

Comment:
Any reason to not pick up icon files automatically, like how Django does it with templates?